tequilapi/endpoints/list_identities.go

```diff
@@ -0,0 +1,30 @@
+package endpoints
+
+import (
+	"net/http"
+
+	"github.com/julienschmidt/httprouter"
+	"github.com/mysterium/node/identity"
+	"github.com/mysterium/node/service_discovery/dto"
+	"github.com/mysterium/node/tequilapi/utils"
+)
+
+type identityList struct {
+	Identities []dto.Identity `json:"identities"`
+}
+
+type listIdentitiesEndpoint struct {
+	listIdentities func() []dto.Identity
+}
+
+func ListIdentitiesEndpointFactory(idm *identity.IdentityManager) *listIdentitiesEndpoint {
+	return &listIdentitiesEndpoint{listIdentities: idm.GetIdentities}
+}
+
+func (lsid *listIdentitiesEndpoint) ListIdentities(writer http.ResponseWriter, request *http.Request, params httprouter.Params) {
+	list := identityList{
+		Identities: lsid.listIdentities(),
```

Comment:
I miss DTO layer which is mapping internal structure -> to API structures

Reply:
please check `service_discovery/dto/identity.go` if it's what you meant to say

Reply:
REST DTOs should be nest to REST service. See example of Healthcheck endpoint